src/Annotation.cpp

```diff
@@ -0,0 +1,500 @@
+
+  if (rawMethodsStringSize == 0) {
+    logout("Method Args Error 0 on '"
+           << rawMethodsString << "': string must not be empty") return false;
+  }
+
+  if (!std::isalnum(rawMethodsString[0])) {
+    logout("Method Args Error 1 on '"
+           << rawMethodsString
+           << "': first character of string must be alphanumeric") return false;
+  }
+
+  for (int i = 0; i < rawMethodsStringSize; i++) {
+    if (rawMethodsString[i] == ',') {
+      if ((i == rawMethodsStringSize - 1) || rawMethodsString[i + 1] == ',') {
+        logout("Method Args Error 2 on '"
+               << rawMethodsString
+               << "': comma must not be last character and a comma cannot "
+                  "follow a comma") return false;
+      }
+    } else if (rawMethodsString[i] == ' ') {
+      if ((i == rawMethodsStringSize - 1) || rawMethodsString[i + 1] == ' ' ||
+          rawMethodsString[i + 1] == ',') {
+        logout("Method Args Error 3 on '"
+               << rawMethodsString
+               << "': space must be followed by alnum and cannot be last char, "
+                  "another space, or a comma") return false;
+      }
+    } else if (std::isalnum(rawMethodsString[i])) {
+      if ((i != rawMethodsStringSize - 1) && rawMethodsString[i + 1] == ' ') {
+        logout(
+            "Method Args Error 4 on '"
+            << rawMethodsString
+            << "': alnum character cannot be followed by space") return false;
+      }
+    }
+  }
+  return true;
+}
+
+bool rawStringIsCorrectlyFormatted(const std::string &rawAnnotationString) {
+  std::vector<std::string> chunks =
+      dataflow::splitString(rawAnnotationString, ' ');
+
+  if (chunks.size() < 8) {
+    logout("Annotation String Error 0: malformed string '"
+           << rawAnnotationString
+           << "' potentially missing argument or string is not annotation "
+              "related at all") return false;
+  }
+  // chunks.size() >= 8
+
+  if (chunks[0] != "TOOL_CHECKER") {
+    logout("Annotation String Error 1: invalid annotation '"
+           << rawAnnotationString
+           << "' does not start with 'TOOL_CHECKER' but starts with '"
+           << chunks[0]
+           << "'; string may not be a relevant annotation of the "
+              "analysis") return false;
+  }
+  // chunks[0] == "TOOL_CHECKER"
+
+  if (chunks[1] != "Calls" && chunks[1] != "MustCall") {
+    logout("Annotation String Error 2: invalid annotation '"
+           << rawAnnotationString << "' annotation type '" << chunks[1]
+           << "' is unknown; expected Calls or MustCall") return false;
+  }
+  // chunks[1] == "Calls" or chunks[1] == "MustCall"
+
+  if (chunks[2] != "target") {
+    logout("Annotation String Error 3: invalid annotation '"
+           << rawAnnotationString
+           << "' 'target' keyword missing or inappropriately placed; '"
+           << chunks[2] << "' found") return false;
+  }
+  // chunks[2] == "target"
```

Comment:
What's the purpose of the "target =" keyword? If it's required that the target is next, could we simply remove it and check the next item?

One reason I might imagine you've included it is to make the annotations more readable. That's a good reason, but if so I'd argue that we should permit the "target" section and the "methods" section in either order.

Reply:
It's more for readability on our end. But I don't think we should focus on being able to change the order of "target" and "methods" as we plan in the long run for the user only needing to be able to specify the methods.